pkg/retriever/selectorutils/selectorutils.go

```diff
@@ -0,0 +1,76 @@
+package selectorutils
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/node/basicnode"
+	"github.com/ipld/go-ipld-prime/traversal/selector"
+	"github.com/ipld/go-ipld-prime/traversal/selector/builder"
+)
+
+// Kind of a combination of these two, with some additional magic sprinkles:
+// https://github.com/ipld/go-ipld-selector-text-lite/blob/master/parser.go
+// https://github.com/ipfs/go-unixfsnode/blob/9cc15a4574f13f434f2da2cd1afb9de7d0bb3979/signaling.go#L23
+// TODO: upstream this, probably to go-unixfsnode
+
+// PathToSelector converts a standard IPLD path to a selector that explores the
+// whole UnixFS path (inclusive), and if 'full' is true, the complete DAG at its
+// termination, or if not true, only the UnixFS node at its termination (which
+// may be an entire sharded directory or file).
+//
+// Path is optional, but if supplied it must start with a '/'.
+//
+// This selector does _not_ match, only explore, so it's useful for traversals
+// where block loads are important, not where the matcher visitor callback is
+// important.
+func UnixfsPathToSelector(path string, full bool) (ipld.Node, error) {
+	if len(path) > 0 && path[0] != '/' {
+		return nil, fmt.Errorf("path must start with /")
+	}
+
+	ssb := builder.NewSelectorSpecBuilder(basicnode.Prototype.Any)
+	var ss builder.SelectorSpec
+	if full {
+		// ExploreAllRecursively
+		ss = ssb.ExploreRecursive(
+			selector.RecursionLimitNone(),
+			ssb.ExploreAll(ssb.ExploreRecursiveEdge()),
+		)
+	} else {
+		// ExploreAll only this node, interpreted as unixfs-preload, which will
+		// load sharded files and sharded directories, not go no further.
+		ss = ssb.ExploreInterpretAs("unixfs-preload", ssb.ExploreRecursive(
+			selector.RecursionLimitDepth(0),
+			ssb.ExploreAll(ssb.ExploreRecursiveEdge()),
+		))
```

Comment:
Is there a better option here for this shallow selector? I could use just use a `ssb.ExploreInterpretAs("unixfs-preload", ssb.Matcher())` but I didn't really want to get matchers involved. But recursion depth 0 seems a bit overkill for just "explore the current node, interpreted as unixfs-preload".

/cc @willscott

Reply:
I would use matcher, honestly.